src/contracts/pods/EigenPodManager.sol

```diff
@@ -142,113 +147,41 @@ contract EigenPodManager is
     }
 
     /**
-     * @notice Called by a podOwner to queue a withdrawal of some (or all) of their virtual beacon chain ETH shares.
-     * @param amountWei The amount of ETH to withdraw.
-     * @param withdrawer The address that can complete the withdrawal and receive the withdrawn funds.
+     * @notice Used by the DelegationManager to remove a pod owner's shares while they're in the withdrawal queue.
+     * Simply decreases the `podOwner`'s shares by `shares`, reverting if `shares` exceeds `podOwnerShares[podOwner]`.
      */
-    function queueWithdrawal(
-        uint256 amountWei,
-        address withdrawer
-    )
-        external
-        onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH)
-        onlyNotFrozen(msg.sender)
-        nonReentrant
-        returns (bytes32)
-    {
-        return _queueWithdrawal(msg.sender, amountWei, withdrawer);
+    function removeShares(
+        address podOwner, 
+        uint256 shares
+    ) external onlyDelegationManager {
+        require(shares <= podOwnerShares[podOwner], "EigenPodManager.removeShares: shares amount too high");
```

Comment:
we need to get rid of this check now?

Reply:
no, it's enforcing that a staker can't create a share deficit as the result of a withdrawal

Reply:
I think this is clearer in now that https://github.com/Layr-Labs/eigenlayer-contracts/pull/227 is merged. Please resolve if it's well-documented / clear now; otherwise would really appreciate if you can add to the documentation here!